### DISCUSSION TITLE
Render Long list with minimum proccess usage

### DISCUSSION THREAD

**Discussion by njfamirm** [lit/lit#4802] [Q&A]
Hi

I want render a long list of message bubble:

```json
"1": {
"meta": {
    "id": 1,
    "rev": 1,
    "created": 1724826371217,
    "updated": 1724826371217
},
"data": {
    "from": "user",
    "text": "Hi",
    "type": "text"
}
},
"2": {
"meta": {
    "id": 2,
    "rev": 1,
    "created": 1724826394092,
    "updated": 1724826394092
},
"data": {
    "from": "support",
    "text": "Hi",
    "type": "text"
}
},
```

I'm use repeat for prevent paint extra part and that's work fine, but the render bubble function is called for all message, If there is no change, how can prevent call render function if updated time not changed?

**Comment by sorvell** [comment]
>  the render bubble function is called for all message, If there is no change, how can prevent call render function if updated time not change

It's not clear what you're doing and what you expect to happen. A reproduction of the issue in the lit [playground](https://lit.dev/playground/) would be ideal.

In general, however, the best to render long lists efficiently is to use [@lit-labs/virtualizer](https://github.com/lit/lit/tree/main/packages/labs/virtualizer#readme).
